src/App.jsx

```diff
@@ -13,34 +15,102 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+const REVERSE_SORT = 'reverse';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPrepareGoods(goods, sortBy, reverse) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortBy) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SORT_BY_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+        case REVERSE_SORT:
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (reverse === REVERSE_SORT) {
+    return preparedGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [typeSort, setTypeSort] = useState('');
+  const [reverse, setReverse] = useState('');
+  const sortedGoods = getPrepareGoods(goodsFromServer, typeSort, reverse);
+
+  const clearSort = () => {
+    setTypeSort('');
+  };
+
+  const clearReverse = () => {
+    setReverse('');
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': typeSort !== SORT_BY_ALPHABET,
+          })}
+          onClick={() => setTypeSort(SORT_BY_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': typeSort !== SORT_BY_LENGTH,
+          })}
+          onClick={() => setTypeSort(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': reverse !== REVERSE_SORT,
+          })}
+          onClick={
+            reverse === REVERSE_SORT
+              ? () => clearReverse()
+              : () => setReverse(REVERSE_SORT)
```

Comment:
As reverse(isReversed) is a boolean now
```suggestion
             setReverse(!reverse)
```